src/utils/daos_hdlr.c

```diff
@@ -1344,9 +1376,51 @@ fs_copy(struct cmd_args_s *ap,
 		if (rc == 0)
 			num->num_dirs++;
 		break;
+	case S_IFLNK:
+		/* Similar to cp command to dereference one time*/
+		D_ALLOC(symlink_value, DFS_MAX_PATH + 1);
+		if (symlink_value == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+		rc = fs_query_symlink(ap, src_file_dfs, src_path, symlink_value);
+		if (rc)
+			goto out;
+		/* To determine parent dir */
+		src_parent_path = dirname((char *)src_path);
```

Comment:
[defect] dirname() might modify src_path, normally we'd better alloc a src_path dup and pass it, see example in client/dfs/duns.c

Reply:
I was aware of dirname() changes src_path. Looks like we do not need it anymore here.

Reply:
Yup, caller did not try to access it after this functon return, but this might cause some strange problem in the future if we try to use src_path after this function, and not easy to notice, i am fine with this for now.

Reply:
OK. I will make a duplicate string to be safer. Thank you!